src/sentry/models/user.py

```diff
@@ -492,6 +492,10 @@ def handle_async_deletion(
         pass
 
     def handle_async_replication(self, region_name: str, shard_identifier: int) -> None:
+        from sentry.hybridcloud.rpc.services.caching import region_caching_service
+        from sentry.services.hybrid_cloud.user.service import get_user
+
+        region_caching_service.clear_key(key=get_user.key_from(self.id), region_name=region_name)
```

Comment:
Should the cache also clear on async_deletion?

Reply:
I suppose we can do that, yeah.